pytng/pytng.pyx

```diff
@@ -80,7 +122,7 @@ cdef class MemoryWrapper:
     # holds a pointer to C allocated memory, deals with malloc&free
     # based on:
     # https://gist.github.com/GaelVaroquaux/1249305/ac4f4190c26110fe2791a1e7a6bed9c733b3413f
-    cdef void* ptr
+    cdef void* ptr  #TODO do we want to use std::unique_ptr?
```

Comment:
I was going to say that this is gross and I'm not sure why we wanted it, but then I saw that tng does `realloc` calls, so we need to `malloc` the memory (and not PyMem_Malloc which is Pythons' memory manager call).

@hmacdope does tng still do the reallocs if a buffer isn't large enough?

I don't think we'll need unique_ptr as nobody else ever gets this pointer at the C level.  At the Python level, the `MemoryWrapper` class is reference counted by Python and `__dealloc__` called when it goes out of scope, so it's a little like a unique_ptr again.

Reply:
Noted on the unique ptr, on the reallocs i'm not sure but i can look in the src files.